File: solutions/0169-majority-element/solution.java
class Solution {
    public int majorityElement(int[] nums) {
        int votes = 0;
        int majorElement = -1;
        for (int i = 0; i < nums.length; i++) {
            if(votes==0) majorElement = nums[i];
            if(nums[i]==majorElement){
                votes++;
            }
            else votes--;
        }
        return majorElement;
    }
}

File: solutions/0169-majority-element/solution.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0;
        int ele = nums[0];
        for(int i = 0;i<nums.size();i++){
            if(ele==nums[i]){
                count++;
            }
            else count--;
            if(count<0){
                ele = nums[i];
                count = 1;
            }
        }
        return ele;
    }
};
